### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by jmgq**
### What

Initial setup of this new API. This code was auto-generated via the DP project generation tool.

### How to review

When reviewing, please remember that this is just a skeleton of an app, and the code will be tided up at a later stage. We need this so that we can start building upon this foundation. If you have any advice or gotchas to be aware of in the future, please let us know.

### Who can review

Anyone.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  funlen:
+  # lines: 100
+  # statements: 100
+
+linters:
+  # please, do not use `enable-all`: it's deprecated and will be removed soon.
+  # inverted configuration with `enable-all` and `disable` is not scalable during updates of golangci-lint
+  disable-all: true
+  enable:
+    - depguard
```
Comment (lindenmckenzie) [inline .golangci.yml:53]: Would remove depguard.

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        - gocyclo
+        - errcheck
+        - dupl
+        - gosec
+  new: false
+
+# golangci.com configuration
+# https://github.com/golangci/golangci/wiki/Configuration
+service:
+  golangci-lint-version: 1.50.x # use the fixed version to not introduce new linters unexpectedly
```
Comment (lindenmckenzie) [inline .golangci.yml:90]: version needs updating to 1.55

File: ci/build.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+---
+
+platform: linux
+
+image_resource:
+  type: docker-image
+  source:
+    repository: golang
+    tag: 1.21.5
```
Comment (lindenmckenzie) [inline ci/build.yml:9]: think this wants to be `1.21.5-bullseye`?

File: ci/component.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+---
+
+platform: linux
+
+image_resource:
+  type: docker-image
+  source:
+    repository: golang
+    tag: 1.21.5
```
Comment (lindenmckenzie) [inline ci/component.yml:9]: think this wants to be `1.21.5-bullseye`?

File: ci/lint.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+---
+
+platform: linux
+
+image_resource:
+  type: docker-image
+  source:
+    repository: golangci/golangci-lint
+    tag: v1.50
```
Comment (lindenmckenzie) [inline ci/lint.yml:9]: v1.55

File: ci/unit.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+---
+
+platform: linux
+
+image_resource:
+  type: docker-image
+  source:
+    repository: golang
+    tag: 1.21.5
```
Comment (lindenmckenzie) [inline ci/unit.yml:9]: think this wants to be `1.21.5-bullseye`?

File: swagger.yaml [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        description: "The last health check date and time of the external service"
+        example: "2020-06-11T11:49:50.330089Z"
+      last_success:
+        type: string
+        description: "The last successful health check date and time of the external service"
+        example: "2020-06-11T11:49:50.330089Z"
+      last_failure:
+        type: string
+        description: "The last failed health check date and time of the external service"
+        example: "2019-09-22T11:48:51.0000001Z"
```
Comment (lindenmckenzie) [inline swagger.yaml:130]: Should end with new line

**Review by lindenmckenzie — COMMENTED** [review]
Few comments - additionally, I think the concourse pipeline should be setup before going any further. Could be wrong on that though.

**Review by lindenmckenzie — DISMISSED** [review]
Approved subject to pipelines being setup and tests passing.

**Review by melsner11Ons — APPROVED** [review]

**Review by lindenmckenzie — APPROVED** [review]

**Pull request merged**